### PR TITLE
docs: mention `git pack-refs` for co-located repos

### PR DESCRIPTION
**Pull request by ilyagr**
As suggested by @yuja in
https://github.com/martinvonz/jj/issues/1841#issuecomment-1720451152

### CODE REVIEW

File: docs/git-compatibility.md [hunk 1]
```diff
@@ -128,6 +128,11 @@ repos may require you to deal with more involved Jujutsu and Git concepts.
   it because they automatically run `git fetch` in the background from time to
   time.
 
+* In co-located repos with a very large number of branches or other refs, `jj`
+  commands can get noticeably slower because of the automatic `jj git import`
+  and `jj git export` executed on each command. This can be mitigated by
+  occasionally running `git pack-refs` to speed up the import and export.
```
[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/git-compatibility.md:134]: Thanks, I had no idea!

Would it be good enough to recommend `git pack-refs --all` every time, just to make the docs a bit briefer/simpler? According to the help, that seems to do a bit of extraneous work, but no real harm.

I was also wondering whether to recommend `git maintenance run` instead, but it seems tricky to figure out how and whether that packs refs.

[the remark this answers is not in the document]
Comment (ilyagr) [reply docs/git-compatibility.md]: In fact, from `git help pack-refs`, it seems that `pack-refs --all` also packs jj's internal refs (which jj creates prolifically), and a simple `pack-refs` might not.

File: docs/git-compatibility.md [hunk 2]
```diff
@@ -128,6 +128,11 @@ repos may require you to deal with more involved Jujutsu and Git concepts.
   it because they automatically run `git fetch` in the background from time to
   time.
 
+* In co-located repos with a very large number of branches or other refs, `jj`
+  commands can get noticeably slower because of the automatic `jj git import`
+  and `jj git export` executed on each command. This can be mitigated by
+  occasionally running `git pack-refs --all` to speed up the import and export.
```
Comment (yuja) [inline docs/git-compatibility.md:134]: nit: I think "export" wouldn't speed up by packing refs

**Review by yuja — APPROVED** [review]

File: docs/git-compatibility.md (hunk 2 above)
Comment (ilyagr) [reply docs/git-compatibility.md]: Good point! Fixed.

**Pull request merged**